lua/oil/columns.lua

```diff
@@ -168,7 +168,14 @@ if icon_provider then
       if meta and meta.display_name then
         name = meta.display_name
       end
-      local icon, hl = icon_provider(field_type, name, conf)
+
+      local ft = ""
+      if field_type == "file" then
+        local firstline = vim.fn.readfile(name, '', 1)
```

Comment:
@stevearc, I have searched, and failed to find how to get absolute filepaths, or the current directory in the context of this function. Do you know how I can get it?

Reply:
The render function takes a bufnr as the third argument
https://github.com/stevearc/oil.nvim/blob/685cdb4ffa74473d75a1b97451f8654ceeab0f4a/lua/oil/columns.lua#L15
You could parse the buffer name to get the parent dir like we do here
https://github.com/stevearc/oil.nvim/blob/685cdb4ffa74473d75a1b97451f8654ceeab0f4a/lua/oil/adapters/files.lua#L506-L507

Reply:
Awesome! Thanks for this!

Done in https://github.com/stevearc/oil.nvim/pull/618/commits/e2ad451f57c5cfa82ef7bded0b44f473e98a2919